vic/drivers/shared_all/src/put_data.c

```diff
@@ -580,17 +554,25 @@ put_data(all_vars_struct   *all_vars,
     }
     storage += out_data[OUT_SWE].data[0] + out_data[OUT_SNOW_CANOPY].data[0] +
                out_data[OUT_WDEW].data[0] + out_data[OUT_SURFSTOR].data[0];
-    out_data[OUT_WATER_ERROR].data[0] = calc_water_balance_error(rec, inflow,
-                                                                 outflow,
-                                                                 storage);
+    if (rec > 0) {
+        out_data[OUT_WATER_ERROR].data[0] = calc_water_balance_error(inflow,
+                                                                     outflow,
+                                                                     storage,
+                                                                     save_data->total_moist_storage);
+    }
+    else {
+        out_data[OUT_WATER_ERROR].data[0] = 0.;
+    }
+
+    // Store total storage for next timestep
+    save_data->total_moist_storage = storage;
 
     /********************
        Check Energy Balance
     ********************/
     if (options.FULL_ENERGY) {
         out_data[OUT_ENERGY_ERROR].data[0] = \
-            calc_energy_balance_error(rec,
-                                      out_data[OUT_SWNET].data[0] +
+            calc_energy_balance_error(out_data[OUT_SWNET].data[0] +
```

Comment:
Shouldn't there be an  `OUT_ENERGY_ERROR`  initialization for `rec < 0 ` like for  `OUT_WATER_ERROR`  in lines 563-565?